das/metrics.go

```diff
@@ -110,29 +123,43 @@ func (d *DASer) InitMetrics() error {
 	return nil
 }
 
+// observeSample records the time it took to sample a header +
+// the amount of sampled contiguous headers
 func (m *metrics) observeSample(ctx context.Context, h *header.ExtendedHeader, sampleTime time.Duration, err error) {
 	if m == nil {
 		return
 	}
 	m.sampleTime.Record(ctx, sampleTime.Seconds(),
 		attribute.Bool("failed", err != nil),
-		attribute.Int("header_width", len(h.DAH.RowsRoots)))
+	)
+
 	m.sampled.Add(ctx, 1,
 		attribute.Bool("failed", err != nil),
-		attribute.Int("header_width", len(h.DAH.RowsRoots)))
+	)
+
 	atomic.StoreInt64(&m.lastSampledTS, time.Now().UTC().Unix())
 }
 
+// observeGetHeader records the time it took to get a header from the header store.
 func (m *metrics) observeGetHeader(ctx context.Context, d time.Duration) {
 	if m == nil {
 		return
 	}
 	m.getHeaderTime.Record(ctx, d.Seconds())
 }
 
+// observeNewHead records the network head.
 func (m *metrics) observeNewHead(ctx context.Context) {
 	if m == nil {
 		return
 	}
 	m.newHead.Add(ctx, 1)
 }
+
+// recordTotalSampled records the total amount of sampled headers.
+func (m *metrics) recordTotalSampled(ctx context.Context, n int64) {
+	if m == nil {
+		return
+	}
+	m.totalSampled.Add(ctx, n)
```

Comment:
Oh man, the counter could not be used like that, you need gauge. I won't be long until you observe sampled percentage to be > 100%